lapack/testlapack/dgeql2.go

```diff
@@ -71,7 +71,7 @@ func Dgeql2Test(t *testing.T, impl Dgeql2er) {
 			copy(qTmp.Data, q.Data)
 			blas64.Gemm(blas.NoTrans, blas.NoTrans, 1, h, qTmp, 0, q)
 		}
-		if !isOrthonormal(q) {
+		if !isOrthogonal(q) {
```

Comment:
perhaps we should update the error message to say `orthogonal` instead of `orthonormal` ?

Reply:
Thanks, I missed that. I will change the error message.

Reply:
Doing this I noticed that `isOrthogonal` can be also used in `Dsytd2Test`. It has also brought to my attention the existence of `hasOrthonormalColumns`. All in all, I smuggled in some additional changes in the last 4 commits.

Please take a look.